pkg/install/deploystorage.go

```diff
@@ -212,43 +181,8 @@ func (i *Installer) deployStorageTemplate(ctx context.Context, installConfig *in
 		},
 	}
 
-	if os.Getenv("RP_MODE") == "" {
-		t.Resources = append(t.Resources, &arm.Resource{
-			Resource: &mgmtauthorization.DenyAssignment{
-				Name: to.StringPtr("[guid(resourceGroup().id, 'ARO cluster resource group deny assignment')]"),
-				Type: to.StringPtr("Microsoft.Authorization/denyAssignments"),
-				DenyAssignmentProperties: &mgmtauthorization.DenyAssignmentProperties{
-					DenyAssignmentName: to.StringPtr("[guid(resourceGroup().id, 'ARO cluster resource group deny assignment')]"),
-					Permissions: &[]mgmtauthorization.DenyAssignmentPermission{
-						{
-							Actions: &[]string{
-								"*/action",
-								"*/delete",
-								"*/write",
-							},
-							NotActions: &[]string{
-								"Microsoft.Network/networkSecurityGroups/join/action",
-							},
-						},
-					},
-					Scope: &i.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID,
-					Principals: &[]mgmtauthorization.Principal{
-						{
-							ID:   to.StringPtr("00000000-0000-0000-0000-000000000000"),
-							Type: to.StringPtr("SystemDefined"),
-						},
-					},
-					ExcludePrincipals: &[]mgmtauthorization.Principal{
-						{
-							ID:   &clusterSPObjectID,
-							Type: to.StringPtr("ServicePrincipal"),
-						},
-					},
-					IsSystemProtected: to.BoolPtr(true),
-				},
-			},
-			APIVersion: azureclient.APIVersions["Microsoft.Authorization/denyAssignments"],
-		})
```

Comment:
moved to a separate function, `denyAssignments`, since it's now needed for both `deployStorageTemplate` and `deploySnapshotUpgradeTemplate`